src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/_params.py

```diff
@@ -93,18 +93,27 @@ def load_arguments(self, _):
         c.argument('vault_name', vault_name_type)
         c.argument('vm', help='Name or ID of the Virtual Machine to be protected.')
         c.argument('policy_name', policy_name_type)
-        c.argument('container_name', container_name_type)
-        c.argument('item_name', item_name_type)
+
+    for command in ['backup-now', 'disable']:
+        with self.argument_context('backup protection ' + command) as c:
+            c.argument('container_name', container_name_type)
+            c.argument('item_name', item_name_type)
+
+    with self.argument_context('backup protection backup-now') as c:
         c.argument('retain_until', type=datetime_type, help='The date until which this backed up copy will be available for retrieval, in UTC (d-m-Y).')
-        c.argument('delete_backup_data', arg_type=get_three_state_flag(), help='Option to delete existing backed up data in the Recovery services vault.')
 
+    with self.argument_context('backup protection disable') as c:
+        c.argument('delete_backup_data', arg_type=get_three_state_flag(), help='Option to delete existing backed up data in the Recovery services vault.')
     # Restore
     with self.argument_context('backup restore') as c:
         c.argument('vault_name', vault_name_type)
         c.argument('container_name', container_name_type)
         c.argument('item_name', item_name_type)
         c.argument('rp_name', rp_name_type)
+
+    with self.argument_context('backup restore restore-disks') as c:
         c.argument('storage_account', help='Name or ID of the storge account to which disks are restored.')
+        c.argument('restore_disks_to_this_storage_account', arg_type=get_three_state_flag(), help='Use this flag when you want disks to be restored to the given storage account. When not specified, disks will be restored to their original storage accounts. Default: false.')
```

Comment:
Did this command have a storage account parameter before?  if so, what did it do?  I don't understand the need for the flag.  It seems this could be simplified so that if the user supplies a storage account, restore to that. If they don't, restore to the original.

Reply:
Hi @tjprescott, when the user asks for restore disks, there are 2 pieces of important information which needs to be given back to the user.
1)	The configuration of the backed up VM (in the form of VMConfig.json)
2)	The data of the backed up VM i.e., disks of the VM.

We always need a storage account to push the configuration file. Previously both the artifacts were restored to the storage account specified by the user in the restore command. Now, with the new flag, users have the capability of deciding the destination of the restored disks. When this flag is set to true, the disks will be restored to the storage account provided in the command (via --storage-account). When the flag is set to false, the disks will be restored to the original storage accounts of the disks themselves.

Hope I was able to make myself clear. Please let me know if you need any more info for this review.

Reply:
So the default then would be to restore to the orginal storage account, unless otherwise specified? In that case, since you have a breaking change anyways (in the sense that the default behavior of the command has changed) then why not simply have a --source-storage and --dest-storage parameter where, if --dest-storage is omitted, it restores to the original storage accounts?

If nothing else, there should be a better name for the flag than `--restore-disks-to-this-storage-account` which I would not even interpret as a flag but expect it to take a storage account name or ID.  So the name of the flag is overly verbose and misleading.

Reply:
Our PMs @pvrk and @trinadhkotturu can answer this.

Reply:
This behavior of restoring to original storage accounts is only applicable for customers who are upgraded to our new stack. This is in currently private preview and will go GA next month. So, when we release this feature, it will not be a breaking change for existing customers as since we can't restore disks to original storage accounts and we restore them to storage account specified. So no behavior  change for them. For customers who are enrolled in private preview and who are going to upgrade to new stack, anyway, we tell that this is the behavior they are getting. So, they are aware of the change. The only optimization we are doing here is to make the best user experience as the default option so that customer need not specify additional flags. 
Reg verbosity of the flag, --restore-to-this-storageAcct will this be fine?

Reply:
As I think more, --restore-to-staging-storage-account makes more sense as in a way that new storage account is a staging location.